sql/2024/fonts/common.sql

```diff
@@ -0,0 +1,94 @@
+-- Normalize a name. Used in FAMILY.
+CREATE TEMPORARY FUNCTION FAMILY_INNER(name STRING) AS (
+  IF(LENGTH(TRIM(name)) < 3, NULL, NULLIF(TRIM(name), ''))
+);
+
+-- Extract the family name from a payload.
+CREATE TEMPORARY FUNCTION FAMILY(payload STRING) AS (
+  FAMILY_INNER(JSON_EXTRACT_SCALAR(payload, '$._font_details.names[1]'))
```

Comment:
We should probably pull this from the CSS as in 2022, as it is very common to remove the name table entries for web fonts (i.e. why a lot of the entries in the data are empty).

Reply:
I did not change this aspect. There are queries where the name is taken from the CSS, and there are queries where the name is taken from the naming table. This function is used in the latter queries.

Reply:
More specifically, it is used in the following queries, which are based solely on the font files:

- design/fonts_family_by_foundry.sql,
- design/fonts_family_by_script.sql,
- development/fonts_color_family.sql,
- development/fonts_variable_family.sql, and
- performance/fonts_family_by_service.sql.